packages/flutter_tools/lib/src/commands/build_appbundle.dart

```diff
@@ -77,11 +95,146 @@ class BuildAppBundleCommand extends BuildSubCommand {
+        final BuildResult result = await globals.buildSystem.build(
+          target,
+          env,
+          buildSystemConfig: BuildSystemConfig(
+            resourcePoolSize: null,
+            ),
+          );
+        if (!result.success) {
+          for (final ExceptionMeasurement measurement in result.exceptions.values) {
+            globals.printError('Target ${measurement.target} failed: ${measurement.exception}',
+              stackTrace: measurement.stackTrace
+            );
+          }
+          throwToolExit('Preliminary build for Deferred components setup verification failed.');
+        }
+        // Setup "prebuild" complete
+
+        // Begin setup verification
+        Directory androidDir = env.projectDir.childDirectory('android');
+        Directory outputDir = env.projectDir
+            .childDirectory('build')
+            .childDirectory(kDeferredComponentsTempDirectory);
+        List<String> generatedFiles = <String>[];
+        List<String> modifiedFiles = <String>[];
+        for (DeferredComponent component in deferredComponents) {
+          DeferredComponentAndroidFiles androidFiles = DeferredComponentAndroidFiles(name: component.name, env: env);
+          if (!androidFiles.verifyFilesExist()) {
+            // generate into temp directory
+            generatedFiles.addAll(
+              androidFiles.generateFiles(
+                alternateAndroidDir: outputDir,
+                clearAlternateOutputDir: true,
+              )
+            );
+          }
+        }
+        List<LoadingUnit> generatedLoadingUnits = parseGeneratedLoadingUnits(env);
+        modifiedFiles.addAll(DeferredComponentAndroidFiles.setupAndroidFiles(env, deferredComponents, generatedLoadingUnits, outputDir));
+        // TODO: Make the modified files requirement softer. We cannot always parse user
+        // build.gradle/settings.gradle files accurately so the modified files list may
+        // produce false positives. We should warn here, but not exit.
+        bool fileChangesNeeded = generatedFiles.isNotEmpty || modifiedFiles.isNotEmpty;
+        if (fileChangesNeeded) {
+          print('''
+===========================================================================
+                     Deferred Components Setup Verification
+===========================================================================''');
+          if (generatedFiles.isNotEmpty) {
+            print('  Newly generated android files:');
+            for (String filePath in generatedFiles) {
+              print('    - $filePath');
+            }
+          }
+          if (modifiedFiles.isNotEmpty) {
+            print('\n  Modified android files:');
+            for (String filePath in modifiedFiles) {
+              print('    - $filePath');
+            }
+          }
+          if (generatedFiles.isNotEmpty && modifiedFiles.isNotEmpty) {
+            print('''
+
+The above files have been placed into `build/$kDeferredComponentsTempDirectory`
```

Comment:
Oh I see. Maybe the directory name shouldn't say "example" since it's not really an example/choice. It's forced for the build to pass.